tools/outline_proxy_controller/outline_proxy_controller.cpp

```diff
@@ -303,7 +305,7 @@ void OutlineProxyController::routeThroughOutline(std::string outlineServerIP) {
     // We failed to make a route through outline proxy. We just remove the flag
     // indicating DNS is backed up.
     resetFailRoutingAttempt(OUTLINE_PRIORITY_SET_UP);
-    return;
+    throw std::system_error{ErrorCode::kConfigureSystemProxyFailure};
```

Comment:
Every error here is `kConfigureSystemProxyFailure`. That's not very useful.

The caller should receive the actual error instead, and maybe propagate as kConfigureSystemProxyFailure later, but it's important to know what failed.

This is fine for this PR, but we should think about that.

Reply:
We can also put the message in `system_error`, and it will also be printed to the system logs, user can query that through `journalctl -u outline_proxy_controller --since today`.
`.

Reply:
Sure, but I'd like that to propagate all the way to the UI.